### ISSUE TITLE
Allow configuration of additional allowed URL schemes in user content 

### ISSUE THREAD

**Issue by baylf2000** [vanilla/vanilla#7769]
We run a cluster of gaming servers and would like to utilise Vanilla for our forums, however we have a small issue in that our users often need to post links to allow users to directly connect to their gaming servers.

These links look like "steam://connect/1.2.3.4:1234". These links can be used in any web browser to open the Steam client, run the appropriate game and connect to that server.

Steam (from Valve Corporation Ltd) is the world's largest digital distribution platform for PC games, and almost all PC games are distributed via this platform. See https://en.wikipedia.org/wiki/Steam_(software)

I would like to suggest the addition of "steam://" to the allowed url schemes for Vanilla.

Many thanks.

**Comment by charrondev** [comment]
I definitely agree that it would be nice to allow such links. What I don't want to do is maintain a list of application specific URL schemes.

Would something like a configuration option for additional allowable URL schemes be sufficient? We don't currently have the bandwidth as a team to implement something like this as a priority right now, but if you are able I could point you in the right direction to make a contribution like this.

To summarize:

- Add a configuration option to the `VanillaHtmlFormatter` class. For consistency let's call it `'Garden.Html.CustomUrlSchemes`.
- Merge these the values of this array into the Htmlawed config.
- Due to the performance sensitive natures of this HTML filtering, be sure to implement the config lookup use the `StaticCacheConfigTrait` the `StaticCacheConfigTrait::c()` to lookup the value.

**Comment by baylf2000** [comment]
A configuration option sounds like a great solution. I'm sure there are other url types out there that users may want to add themselves,

I'm far from a professional developer, but I'd be happy to at least take a look at anything you can suggest that might help me make such a contribution.

**Comment by linc** [comment]
I'd be happy to allow `steam://` links by default.
